mpt/impl/imap-mailbox/core/src/main/java/org/apache/james/mpt/imapmailbox/suite/ACLIntegration.java

```diff
@@ -19,245 +19,280 @@
 
 package org.apache.james.mpt.imapmailbox.suite;
 
+import java.util.Locale;
+
 import org.apache.james.mailbox.model.MailboxPath;
 import org.apache.james.mailbox.model.SimpleMailboxACL;
 import org.apache.james.mpt.api.ImapHostSystem;
 import org.apache.james.mpt.imapmailbox.GrantRightsOnHost;
+import org.apache.james.mpt.imapmailbox.ImapTestConstants;
 import org.apache.james.mpt.imapmailbox.MailboxMessageAppender;
-import org.apache.james.mpt.imapmailbox.suite.base.BaseImapProtocol;
+import org.junit.Before;
 import org.junit.Test;
 
-import javax.inject.Inject;
-import java.util.Locale;
-
-public class ACLIntegration extends BaseImapProtocol {
+public abstract class ACLIntegration implements ImapTestConstants {
     public static final String OTHER_USER_NAME = "Boby";
     public static final String OTHER_USER_PASSWORD = "password";
     public static final MailboxPath OTHER_USER_MAILBOX = new MailboxPath("#private", OTHER_USER_NAME, "");
     public static final MailboxPath MY_INBOX = new MailboxPath("#private", USER, "");
 
-    @Inject
-    private static ImapHostSystem system;
-    @Inject
+    protected abstract ImapHostSystem createImapHostSystem();
+    protected abstract GrantRightsOnHost createGrantRightsOnHost();
+    protected abstract MailboxMessageAppender createMailboxMessageAppender();
```

Comment:
Couldn't the MailboxMessageAppender be retrieved from the host system? Idem for GrantRightsOnHost?